File: Lab_9/computer.py
from processor import Processor
from memory import Memory
from graphics_card import GraphicsCard

class Computer:
    def __init__(self, processor=Processor(""), memory=Memory(0), graphics_card=GraphicsCard("")):
        self.processor = processor
        self.memory = memory
        self.graphics_card = graphics_card

    def turn_on(self):
        self.log("Комп'ютер увімкнено.")

    def turn_off(self):
        self.log("Комп'ютер вимкнено.")

    def run_program(self, program):
        self.log(f"Виконується програма: {program}")

    def upgrade_memory(self, additional_memory):
        self.memory.add_capacity(additional_memory)
        self.log(f"Пам'ять розширено на {additional_memory} ГБ.")

    def get_processor_model(self):
        self.log(f"Отримання назви моделі процесора, а саме: {self.processor.get_model()}")
        return self.processor.get_model()

    def get_memory_capacity(self):
        capacity = self.memory.get_capacity()
        self.log(f"Отримання інформації про місткість пам'яті комп'ютера, яка становить: {capacity} ГБ")
        return capacity

    def get_graphics_card_model(self):
        self.log(f"Отримання назви моделі відеокарти, а саме: {self.graphics_card.get_model()}")
        return self.graphics_card.get_model()

    def update_processor_model(self, new_model):
        self.processor.set_model(new_model)
        self.log(f"Модель процесора оновлено на: {new_model}")

    def display_computer_info(self):
        self.log(f"Інформація про комп'ютер:\n"
                 f"Процесор: {self.get_processor_model()}\n"
                 f"Пам'ять: {self.get_memory_capacity()} ГБ\n"
                 f"Графічна карта: {self.get_graphics_card_model()}")

    def log(self, message):
        with open("log.txt", "a") as writer:
            writer.write(message + "\n")

File: Lab_2/src/CI306/Zarytskyi/Lab2/Computer.java
package CI306.Zarytskyi.Lab2;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
/**
 * Клас, який представляє комп'ютер.
 */
public class Computer {
    private Processor processor;
    private Memory memory;
    private GraphicsCard graphicsCard;

    /**
     * Конструктор для створення комп'ютера з параметрами за замовчуванням.
     * Ініціалізує об'єкти Processor, Memory та GraphicsCard з порожніми значеннями.
     */
    public Computer() {
        this.processor = new Processor("");
        this.memory = new Memory(0);
        this.graphicsCard = new GraphicsCard("");
    }

    /**
     * Конструктор для створення комп'ютера з вказаними комплектуючими.
     * @param processor Процесор комп'ютера.
     * @param memory Пам'ять комп'ютера.
     * @param graphicsCard Графічна карта комп'ютера.
     */
    public Computer(Processor processor, Memory memory, GraphicsCard graphicsCard) {
        this.processor = processor;
        this.memory = memory;
        this.graphicsCard = graphicsCard;
    }

    /**
     * Метод для включення комп'ютера.
     */
    public void turnOn() {
        log("Комп'ютер увімкнено.");
    }

    /**
     * Метод для вимкнення комп'ютера.
     */
    public void turnOff() {
        log("Комп'ютер вимкнено.");
    }

    /**
     * Метод для виконання програми на комп'ютері.
     * @param program Назва програми, яку слід виконати.
     */
    public void runProgram(String program) {
        log("Виконується програма: " + program);
    }

    /**
     * Метод для розширення обсягу пам'яті комп'ютера.
     * @param additionalMemory Кількість додаткової пам'яті в гігабайтах.
     */
    public void upgradeMemory(int additionalMemory) {
        memory.addCapacity(additionalMemory);
        log("Пам'ять розширено на " + additionalMemory + " ГБ.");
    }

    /**
     * Метод для отримання моделі процесора комп'ютера.
     * @return Модель процесора.
     */
    public String getProcessorModel() {
    	log("Отримання назву моделі процесора, а саме: " + processor.getModel());
        return processor.getModel();
    }

    /**
     * Метод для отримання обсягу встановленої пам'яті комп'ютера.
     * @return Обсяг пам'яті в гігабайтах.
     */
    public int getMemoryCapacity() {
    	log("Отримання інформації про місткість пам'яті комп'ютера, яка становить: " + memory.getCapacity() + "ГБ");
        return memory.getCapacity();
    }

    /**
     * Метод для отримання моделі графічної карти комп'ютера.
     * @return Модель графічної карти.
     */
    public String getGraphicsCardModel() {
    	log("Отримання назву моделі відеокарти, а саме: " + graphicsCard.getModel());
        return graphicsCard.getModel();
    }

    /**
     * Метод для оновлення моделі процесора комп'ютера.
     * @param newModel Нова модель процесора.
     */
    public void updateProcessorModel(String newModel) {
        processor.setModel(newModel);
        log("Модель процесора оновлено на: " + newModel);
    }

    /**
     * Метод для виведення інформації про комп'ютер у консоль або в файл "log.txt".
     */
    public void displayComputerInfo() {
        log("Інформація про комп'ютер:\n" +
                "Процесор: " + getProcessorModel() + "\n" +
                "Пам'ять: " + getMemoryCapacity() + " ГБ\n" +
                "Графічна карта: " + getGraphicsCardModel());
    }

    /**
     * Метод для виведення message у файл "log.txt".
     */
    private void log(String message) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("log.txt", true))) {
            writer.write(message + "\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

